src/pages/settings/user/components/SmsVerificationModal.tsx

```diff
@@ -0,0 +1,121 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Button, InputField } from '@invoiceninja/forms';
+import { Modal } from 'components/Modal';
+import { Dispatch, SetStateAction, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface Props {
+  visible: boolean;
+  setVisible: Dispatch<SetStateAction<boolean>>;
+  resendSmsCode: () => void;
+  verifyPhoneNumber: (code: string) => void;
+}
+
+export function SmsVerificationModal(props: Props) {
+  const [t] = useTranslation();
+
+  const digitFieldsBox = useRef<HTMLDivElement>(null);
+
+  const digitFields = [
+    'digit1',
+    'digit2',
+    'digit3',
+    'digit4',
+    'digit5',
+    'digit6',
+  ];
+
+  const initialCodeValue = ['', '', '', '', '', ''];
+
+  const [code, setCode] = useState<string[]>(initialCodeValue);
+
+  const [isCodeEntered, setIsCodeEntered] = useState<boolean>(false);
+
+  const handleChangeValue = (value: string, fieldIndex: number) => {
+    const updatedCode = code.map((currentValue, index) =>
+      index === fieldIndex ? value : currentValue
+    );
+
+    const isCodeFullyEntered = updatedCode.every((value) => value);
+
+    setIsCodeEntered(isCodeFullyEntered);
+
+    setCode(updatedCode);
+
+    if (value) {
+      if (fieldIndex === digitFields.length - 1) {
+        (
+          digitFieldsBox.current?.children[0].children[0]
+            .children[0] as HTMLInputElement
+        ).focus();
+      } else {
+        (
+          digitFieldsBox.current?.children[fieldIndex + 1].children[0]
+            .children[0] as HTMLInputElement
+        ).focus();
+      }
+    }
+  };
+
+  const handleCodeClear = () => {
+    (
+      digitFieldsBox.current?.children[0].children[0]
+        .children[0] as HTMLInputElement
```

Comment:
I had to access the children twice, because we actually have a section in the InputField component as the first parent, and then a div as the second parent to access the actual input field.